packages/core/lib/use-puck-history.ts

```diff
@@ -37,6 +39,27 @@ export function usePuckHistory({
     }
   };
 
+  const setHistories = (histories: History[]) => {
+    // dispatch the last history index or initial state
+    dispatch({
+      type: "set",
+      state: histories[histories.length - 1]?.data || initialAppState,
+    });
+
+    historyStore.setHistories(histories);
+  };
+
+  const setHistoryIndex = (index: number) => {
+    if (historyStore.histories.length > index) {
+      dispatch({
+        type: "set",
+        state: historyStore.histories[index]?.data || initialAppState,
+      });
+    }
+
+    historyStore.setHistoryIndex(index);
+  };
```

Comment:
Can you add some unit tests to `__tests__/use-puck-history.spec.ts` for this?

Reply:
Added, which caught some things 👍